### PR TITLE
added patch to remove error messages from OpenCV3's python discovery

### PR DESCRIPTION
**Pull request by eggerk**
@mfehr @dymczykm 
swe still compiles and runs after this

### CODE REVIEW

File: fix_python_discovery.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+diff --git cmake/OpenCVDetectPython.cmake b/cmake/OpenCVDetectPython.cmake
+index 6dec76f..fba147a 100644
+--- cmake/OpenCVDetectPython.cmake
++++ cmake/OpenCVDetectPython.cmake
+@@ -177,6 +177,7 @@ if(NOT ${found})
+           execute_process(COMMAND "${_executable}" -c "import os; os.environ['DISTUTILS_USE_SDK']='1'; import numpy.distutils; print(os.pathsep.join(numpy.distutils.misc_util.get_numpy_include_dirs()))"
+                           RESULT_VARIABLE _numpy_process
+                           OUTPUT_VARIABLE _numpy_include_dirs
++                          ERROR_QUIET
```
Comment (mfehr) [inline fix_python_discovery.patch:15]: Nice! Smallest patch ever :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,10 +25,11 @@ ExternalProject_Add(opencv3_src
   UPDATE_COMMAND ""
   SOURCE_DIR ${OPENCV_SRC_PATH}
   BINARY_DIR opencv3_build
+  PATCH_COMMAND patch -p0 < ${CMAKE_SOURCE_DIR}/fix_python_discovery.patch
   CONFIGURE_COMMAND cd ../${OPENCV_SRC_PATH} && cmake
     -DOPENCV_EXTRA_MODULES_PATH=../${CONTRIB_NAME}_src/modules .
     -DWITH_TBB=ON
-    -DBUILD_opencv_python2=OFF
+    -DBUILD_opencv_python2=ON
```
Comment (mfehr) [inline CMakeLists.txt:32]: Probably a stupid question, but could you briefly elaborate why this change is necessary, i.e. why this was turned off before?

Comment (eggerk) [reply CMakeLists.txt:32]: We turned it off to see if this would remove the numpy error message. We can reenable python support now because that wasn't the problem

**Comment by eggerk** [comment]
@mfehr @dymczykm what's the state? can we merge this?

**Review by dymczykm — APPROVED** [review]
yes, let's merge it

lgtm
